### ISSUE TITLE
Error in run-android script on Windows

### ISSUE THREAD

**Issue by Jeopardye** [electrode-io/electrode-native#548]
Howdy. I ran into an issue with run-android script on my Windows machine. I get the following error:

Generating Container[fillContainerHull] Something went wrong: Error: Command failed: C:\Users\myusername\.ern\versions\0.12.0\node_modules\.bin\
yarn add @apps\react-native-ourapp-navigation-api@1.0.5 --ignore-engines --non-interactive --exact  
error An unexpected error occurred: "https://registry.yarnpkg.com/@apps/react-native-ourapp-navigation-api: Not found".

[generateContainer] Something went wrong. Aborting Container Generation
× Command failed: C:\Users\myusername\.ern\versions\0.12.0\node_modules\.bin\yarn add @apps\react-native-ourapp-navigation-api@1.0.5 --ignore-engines
 --non-interactive --exact
error An unexpected error occurred: "https://registry.yarnpkg.com/@apps/react-native-ourapp-navigation-api: Not found".

runLocalContainerGen failed: Error: Command failed: C:\Users\myusername\.ern\versions\0.12.0\node_modules\.bin\yarn add @apps\react-native-ourapp-nav
igation-api@1.0.5 --ignore-engines --non-interactive --exact  
error An unexpected error occurred: "https://registry.yarnpkg.com/@apps/react-native-ourapp-navigation-api: Not found".

An error occurred: Command failed: C:\Users\myusername\.ern\versions\0.12.0\node_modules\.bin\yarn add @apps\react-native-ourapp-navigation-api@1.0.5
 --ignore-engines --non-interactive --exact  
error An unexpected error occurred: "https://registry.yarnpkg.com/@apps/react-native-ourapp-navigation-api: Not found".

This time the issue is the `\` char after the @apps-scope in the "yarn add" command. If I switch the `\` to `/` in that command, it runs fine (and uses the right registry).

So this doesn't work:
`yarn add @apps\react-native-ourapp-navigation-api@1.0.5`
This works:
`yarn add @apps/react-native-ourapp-navigation-api@1.0.5`

On a Mac that script runs fine, so I guess it needs some Windows-specific love so that the slash is the right way with custom scopes.

**Comment by Jeopardye** [comment]
I tried to fight my way through the issue by making a hack in YarnCli.js class which will replace the `\` with `/` when it finds the @apps scope prefix, but instead of fixing the issue it gave me the following error:

[fillContainerHull] Something went wrong: Error: Command failed: C:\Users\myusername\.ern\versions\0.12.0\node_modules\.bin\
yarn add @apps/react-native-ourapp-navigation-api@1.0.5 --ignore-engines --non-interactive --exact  
error An unexpected error occurred: "https://registry.yarnpkg.com/@apps%2freact-native-ourapp-navigation-api: Not found".
..........

So there's some encoding thing hidden somewhere that turns the `/` into `%2f`. What's annoying is that that command works fine when I execute it by hand (because the `/` stays as `/`)

**Comment by belemaire** [comment]
@Jeopardye 
Will take a closer look into this today along with two other Windows reported issue. This is a bit weird, don't really know why in the first place ern uses `\` on windows for scoped packages, shouldn't be the case.
Looking into two other reported Windows issues, will probably release a patch version 0.12.2 to fix reported problems on Windows by this evening.

**Comment by belemaire** [comment]
@Jeopardye 
Looked a bit into it, don't really see where this could come from.
Could you please provide us some more info ?
I think we agree that the issue is due to using a `\` to delimit the scope and the package name, vs using `/`, which is the proper way (https://docs.npmjs.com/misc/scope). What I am trying to understand is how this `\` ended up being used as a delimiter in the first place as it does not seem we are doing any replacement in the platform code. 
I am not sure of the way you generate the Container (as there are multiple ways : through runner, or `create-container` command or from a cauldron ...). In any case, the source that refers to this API (either Cauldron, or inside the MiniApp or whatever provided as extra dependency to `create-container`) should use the right delimiter `/` in the first place. Could you confirm that this is the case ? If not and the source is using `\`, that's the issue. Otherwise will have to deep dive a bit more in the code to see why it's getting replaced to `\`.

**Comment by Jeopardye** [comment]
Basically I run the create-container and run-android inside the miniapp folder. The delimiter inside the package.json is `/`, so the right way.

What's interesting is that there are two dependencies in the package.json with that apps-scope, but only the second one misbehaves:
```
"dependencies": {
    "@apps/react-native-ourapp-common": "0.0.56",
    "@apps/react-native-ourapp-navigation-api": "1.0.5",
```

The god-awful hack I made in YarnCli.js within the add-function looks like the following:
```
var path = dependencyPath.toString();
if (path.startsWith("@apps\\")) {
  path = path.replace("@apps\\", "@apps/")
}
cmd = `add ${path} --ignore-engines --non-interactive --exact ${dev ? '--dev' : ''} ${peer ? '--peer' : ''}`;
```
And with that, the script runs fine.

**Comment by belemaire** [comment]
@Jeopardye 
Thanks for the additional details. Indeed the hack is not one to make public, very specific to your scope ;)
Finally found out the root cause of the problem and issued a PR to fix it (https://github.com/electrode-io/electrode-native/pull/557).
We will release `0.12.2` later on today including this fix (still have to complete something else that needs to make it in this patch version). Meanwhile you can keep still using your temporary workaround.
Thanks again for reporting !

**Comment by belemaire** [comment]
FYI, just released Electrode Native [0.12.2](https://github.com/electrode-io/electrode-native/releases/tag/v0.12.2) which includes the fix to this issue.
